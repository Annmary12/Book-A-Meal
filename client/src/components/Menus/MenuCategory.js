import React from 'react';
import PropTypes from 'prop-types';
import MealDropDown from './MealDropDown.js';
import styles from './menus.scss';

const MenuCategory = ({ category }) => (
  <div>
    <h3 className={styles.categoryTitle}>-{category.title}</h3>
    <ul className={styles.mealList}>
      {
        category.meals.map(meal => (
          <li key={meal.id}>
            <MealDropDown meal={meal} />
          </li>
          ))
      }
    </ul>
  </div>
);

MenuCategory.propTypes = {
  category: PropTypes.objectOf(PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.array
  ])).isRequired
};

export default MenuCategory;
