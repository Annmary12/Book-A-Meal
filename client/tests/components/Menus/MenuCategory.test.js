import React from 'react';
import MenuCategory from '../../../src/components/Menus/MenuCategory';
import { wrap } from 'module';

const category = {
  meals: [{
    id: 'test-id',
    title: 'fufu and egusi',
    price: 230,
    img: 'fufu.jpg'
  }]
};

describe('MenuCategory component', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(
      <MenuCategory
        category={category}
      />
    );

    expect(wrapper).toMatchSnapshot();
  });
});