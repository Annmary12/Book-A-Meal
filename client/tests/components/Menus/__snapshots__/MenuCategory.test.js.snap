// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuCategory component should match snapshot 1`] = `
<div>
  <h3
    className="categoryTitle"
  >
    -
  </h3>
  <ul
    className="mealList"
  >
    <li
      key="test-id"
    >
      <MealDropDown
        meal={
          Object {
            "id": "test-id",
            "img": "fufu.jpg",
            "price": 230,
            "title": "fufu and egusi",
          }
        }
      />
    </li>
  </ul>
</div>
`;
